samples/99_zil_interpreter/app/main.rb

```diff
@@ -7,39 +7,106 @@
 
 def tick(args)
   setup(args) if args.tick_count.zero?
-  # TODO:
-  # - Render history
-  # - Collect input
-  # - Call send_input with input when pressing Enter
+  # Render history
+  render_state(args)
+  
+  # Collect input
+  handle_input(args)
+  
+  # Call send_input when pressing Enter
+  if args.inputs.keyboard.key_down.enter && args.state.input != ''
+    send_input(args, args.state.input)
+    args.state.input = ''
+  end
+  
   $gtk.request_quit unless $interpreter.alive?
 end
 
 def setup(args)
+  args.state.text_color = [240, 255, 255]
+  args.state.bg_color = [0, 0, 0]
+  args.state.input = ''
+  args.state.text_history = []
+  args.state.current_line = 0
   context = build_zil_context(args)
   args.state.zil_context = context
   $interpreter = Fiber.new {
     context.globals[:GO].call [], context
   }
   $interpreter.resume # Initial processing until first Fiber.yield
-  process_outputs args, context.outputs # Process welcome message if existing
+  process_outputs args, ' ', context.outputs # Process welcome message if existing
+  context.outputs.clear
 
   # TODO:
   # Add other setup if necessary
 end
 
-# Call this method with the input after pressing enter
+def handle_input(args)
+  if args.inputs.keyboard.key_down.backspace
+    args.state.input.chop!
+  else
+    if args.inputs.text[0]
+      args.state.input << args.inputs.text[0]
```

Comment:
one could think about setting current_line to zero here, too... so basically once you type anything you will definitely see what you type